project_mis_report_budget/models/mis_budget.py

```diff
@@ -9,3 +9,8 @@ class MisBudget(models.Model):
 
     project_id = fields.Many2one('project.project')
     analytic_account_project_id = fields.Many2one('account.analytic.account', related='project_id.analytic_account_id')
+
+    def action_duplicate(self):
+        self.ensure_one()
+        budget_id = self.copy()
+        budget_id.state = 'draft'
```

Comment:
Please better use built-in method for this:

https://github.com/OCA/mis-builder/blob/8a19a30b6afc3da50fac83143fbed0dcf100a3d5/mis_builder_budget/models/mis_budget_abstract.py#L55